core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,18 +53,53 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              condition: [/jpeg$/, /\.jpeg$/],
+            },
+            jpg: {
+              format: 'image/jpg',
+              fileType: /jpg$/,
+              fileExtension: /\.jpg$/,
+            },
+            gif: {
+              format: 'image/gif',
+              fileType: /gif$/,
+              fileExtension: /\.gif$/,
+            },
+            png: {
+              format: 'image/png',
+              fileType: /png$/,
+              fileExtension: /\.png$/,
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              fileType: /svg\+xml$/,
+              fileExtension: /\.svg$/,
+            }
+          };
+          var imageHasValidFormat = true;
+          var imageHasValidExtension = false;
+          for (var i = 0; i < scope.getAllowedImageFormats().length; i++) {
+            var fileType = scope.getAllowedImageFormats()[i];
+            if (!(fileType in imageTypeMapping)) {
+              return 'Unknown image format in allowed image formats.';
+            }
+            imageHasValidFormat = (
+              imageHasValidFormat &&
+              !file.type.match(imageTypeMapping[fileType].format));
+            imageHasValidExtension = (
+              imageHasValidExtension ||
+              (file.type.match(imageTypeMapping[fileType].fileType) &&
+              !file.name.match(imageTypeMapping[fileType].fileExtension)));
+          }
+
+          if (imageHasValidFormat) {
```

Comment:
This is confusing, why return an error when the "imageHasValidFormat" is true? Shouldn't it error when the format is invalid?

Reply:
Done.